xonsh/environ.py

```diff
@@ -2138,6 +2157,28 @@ def register(
                     f"Default value for {name} does not match type specified by validate"
                 )
 
+        # Convert existing variables
+        name_type = builtins.type(name)
+        converted_values = {}
+        for k, val in self._d.items():
```

Comment:
I don't think registering a single env variable should affect any other environment variables.

Reply:
It's not affect any other variables.

Reply:
Yeah, I guess what I am trying to say is that registering should be fast and we should not loop through the env every time a variable is registered.

Reply:
Loop for single variables excluded.